# Beauty Retail Intelligence

Exploratory Data Analysis (EDA)

This notebook explores the Sephora products dataset to understand its structure, evaluate data quality and identify relevant variables before performing business analysis.

## 1. Import Libraries

In [35]:
import pandas as pd
import numpy as np

## 2. Load Dataset

In [36]:
products = pd.read_csv("../data/product_info.csv")

## 3. Initial Exploration

In [37]:
products.head()

,product_id,product_name,brand_id,brand_name,loves_count,rating,reviews,size,variation_type,variation_value,...,online_only,out_of_stock,sephora_exclusive,highlights,primary_category,secondary_category,tertiary_category,child_count,child_max_price,child_min_price
0,P473671,Fragrance Discovery Set,6342,19-69,6320,3.6364,11.0,NaN,NaN,NaN,...,1,0,0,"['Unisex/ Genderless Scent', 'Warm &Spicy Scen...",Fragrance,Value & Gift Sets,Perfume Gift Sets,0,NaN,NaN
1,P473668,La Habana Eau de Parfum,6342,19-69,3827,4.1538,13.0,3.4 oz/ 100 mL,Size + Concentration + Formulation,3.4 oz/ 100 mL,...,1,0,0,"['Unisex/ Genderless Scent', 'Layerable Scent'...",Fragrance,Women,Perfume,2,85.0,30.0
2,P473662,Rainbow Bar Eau de Parfum,6342,19-69,3253,4.2500,16.0,3.4 oz/ 100 mL,Size + Concentration + Formulation,3.4 oz/ 100 mL,...,1,0,0,"['Unisex/ Genderless Scent', 'Layerable Scent'...",Fragrance,Women,Perfume,2,75.0,30.0
3,P473660,Kasbah Eau de Parfum,6342,19-69,3018,4.4762,21.0,3.4 oz/ 100 mL,Size + Concentration + Formulation,3.4 oz/ 100 mL,...,1,0,0,"['Unisex/ Genderless Scent', 'Layerable Scent'...",Fragrance,Women,Perfume,2,75.0,30.0
4,P473658,Purple Haze Eau de Parfum,6342,19-69,2691,3.2308,13.0,3.4 oz/ 100 mL,Size + Concentration + Formulation,3.4 oz/ 100 mL,...,1,0,0,"['Unisex/ Genderless Scent', 'Layerable Scent'...",Fragrance,Women,Perfume,2,75.0,30.0


### Dataset Information

In [38]:
products.info()

<class 'pandas.DataFrame'>
RangeIndex: 8494 entries, 0 to 8493
Data columns (total 27 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   product_id          8494 non-null   str    
 1   product_name        8494 non-null   str    
 2   brand_id            8494 non-null   int64  
 3   brand_name          8494 non-null   str    
 4   loves_count         8494 non-null   int64  
 5   rating              8216 non-null   float64
 6   reviews             8216 non-null   float64
 7   size                6863 non-null   str    
 8   variation_type      7050 non-null   str    
 9   variation_value     6896 non-null   str    
 10  variation_desc      1250 non-null   str    
 11  ingredients         7549 non-null   str    
 12  price_usd           8494 non-null   float64
 13  value_price_usd     451 non-null    float64
 14  sale_price_usd      270 non-null    float64
 15  limited_edition     8494 non-null   int64  
 16  new              

### Descriptive Statistics

In [39]:
products.describe()

,brand_id,loves_count,rating,reviews,price_usd,value_price_usd,sale_price_usd,limited_edition,new,online_only,out_of_stock,sephora_exclusive,child_count,child_max_price,child_min_price
count,8494.000000,8.494000e+03,8216.000000,8216.000000,8494.000000,451.000000,270.000000,8494.000000,8494.000000,8494.000000,8494.000000,8494.000000,8494.000000,2754.000000,2754.000000
mean,5422.440546,2.917957e+04,4.194513,448.545521,51.655595,91.168537,20.207889,0.070285,0.071698,0.219096,0.073699,0.279374,1.631622,53.792023,39.665802
std,1709.595957,6.609212e+04,0.516694,1101.982529,53.669234,79.195631,24.327352,0.255642,0.258002,0.413658,0.261296,0.448718,5.379470,58.765894,38.685720
min,1063.000000,0.000000e+00,1.000000,1.000000,3.000000,0.000000,1.750000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,3.000000,3.000000
25%,5333.000000,3.758000e+03,3.981725,26.000000,25.000000,45.000000,8.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,22.000000,19.000000
50%,6157.500000,9.880000e+03,4.289350,122.000000,35.000000,67.000000,14.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,32.000000,28.000000
75%,6328.000000,2.684125e+04,4.530525,418.000000,58.000000,108.500000,25.000000,0.000000,0.000000,0.000000,0.000000,1.000000,1.000000,59.000000,42.000000
max,8020.000000,1.401068e+06,5.000000,21281.000000,1900.000000,617.000000,320.000000,1.000000,1.000000,1.000000,1.000000,1.000000,105.000000,570.000000,400.000000


### Missing Values

In [40]:
products.isnull().sum()

product_id               0
product_name             0
brand_id                 0
brand_name               0
loves_count              0
rating                 278
reviews                278
size                  1631
variation_type        1444
variation_value       1598
variation_desc        7244
ingredients            945
price_usd                0
value_price_usd       8043
sale_price_usd        8224
limited_edition          0
new                      0
online_only              0
out_of_stock             0
sephora_exclusive        0
highlights            2207
primary_category         0
secondary_category       8
tertiary_category      990
child_count              0
child_max_price       5740
child_min_price       5740
dtype: int64

### Duplicate Records

Check whether the dataset contains duplicated products.

In [41]:
products.duplicated().sum()

np.int64(0)

**Insights**

- No duplicate records were found.
- Each row represents a unique product.
- No duplicate removal is required.

### Product Identifier Validation

Verify whether every product has a unique identifier.

In [42]:
products["product_id"].nunique()

8494

**Insights**

- The dataset contains 8,494 unique product IDs.
- Every product has its own identifier, making it suitable for aggregation and relational analysis.

### Brand Coverage

Explore the diversity of brands included in the dataset.

In [43]:
products["brand_name"].nunique()

304

**Insights**

- The dataset includes 304 different brands.
- This provides enough diversity for meaningful brand-level comparisons.

### Product Categories

Analyze the distribution of products across the main categories.

In [44]:
products["primary_category"].value_counts()

primary_category
Skincare           2420
Makeup             2369
Hair               1464
Fragrance          1432
Bath & Body         405
Mini Size           288
Men                  60
Tools & Brushes      52
Gifts                 4
Name: count, dtype: int64

**Insights**

- Skincare and Makeup dominate the catalog, representing most of the available products.
- Categories such as Gifts and Men contain significantly fewer products.
- This imbalance should be considered when comparing category performance.

### Largest Brands by Number of Products

Identify the brands with the largest product portfolios.

In [45]:
products["brand_name"].value_counts().head(10)

brand_name
SEPHORA COLLECTION         352
CLINIQUE                   179
Dior                       136
tarte                      131
NEST New York              115
Bumble and bumble          110
Kérastase                  108
TOM FORD                   100
Charlotte Tilbury           99
Anastasia Beverly Hills     95
Name: count, dtype: int64

**Insights**

- SEPHORA COLLECTION has the largest catalog in the dataset.
- Several premium brands also maintain broad product portfolios.
- A larger catalog does not necessarily imply better customer engagement, which will be explored later.

### Price Distribution

Understand the overall pricing structure of the catalog.

In [46]:
products["price_usd"].describe()

count    8494.000000
mean       51.655595
std        53.669234
min         3.000000
25%        25.000000
50%        35.000000
75%        58.000000
max      1900.000000
Name: price_usd, dtype: float64

**Insights**

- Most products are priced between $25 and $58.
- The average price is approximately $52.
- Products priced at $1,900 indicate the presence of extreme outliers that should be investigated before visualization.

### Products Without Ratings

Identify products that have not yet received customer ratings.

In [47]:
products[products["rating"].isna()].shape

(278, 27)

**Insights**

- Only 278 products have no rating.
- Missing ratings likely indicate insufficient customer feedback rather than data quality issues.
- These products will be excluded only from analyses that require customer ratings.

### Invalid Prices

Check whether any products have invalid prices.

In [48]:
products[products["price_usd"] <= 0]

,product_id,product_name,brand_id,brand_name,loves_count,rating,reviews,size,variation_type,variation_value,...,online_only,out_of_stock,sephora_exclusive,highlights,primary_category,secondary_category,tertiary_category,child_count,child_max_price,child_min_price


### Rating Validation

Ensure all product ratings fall within the expected range.

In [49]:
products[(products["rating"] < 1) | (products["rating"] > 5)]

,product_id,product_name,brand_id,brand_name,loves_count,rating,reviews,size,variation_type,variation_value,...,online_only,out_of_stock,sephora_exclusive,highlights,primary_category,secondary_category,tertiary_category,child_count,child_max_price,child_min_price


### Review Count Validation

Check for invalid review counts.

In [50]:
products[products["reviews"] < 0]

,product_id,product_name,brand_id,brand_name,loves_count,rating,reviews,size,variation_type,variation_value,...,online_only,out_of_stock,sephora_exclusive,highlights,primary_category,secondary_category,tertiary_category,child_count,child_max_price,child_min_price


### Engagement Validation

Verify that engagement metrics contain valid values.

In [51]:
products[products["loves_count"] < 0]

,product_id,product_name,brand_id,brand_name,loves_count,rating,reviews,size,variation_type,variation_value,...,online_only,out_of_stock,sephora_exclusive,highlights,primary_category,secondary_category,tertiary_category,child_count,child_max_price,child_min_price


### Category Validation

Identify products with missing category information.

In [52]:
products[products["primary_category"].isna()]

,product_id,product_name,brand_id,brand_name,loves_count,rating,reviews,size,variation_type,variation_value,...,online_only,out_of_stock,sephora_exclusive,highlights,primary_category,secondary_category,tertiary_category,child_count,child_max_price,child_min_price


## Data Quality Summary

The dataset presents a high level of data quality. No duplicated records or invalid numerical values were identified. Missing values are primarily associated with optional product attributes (e.g., sale price, variations, ingredients) rather than data quality issues.

Therefore, only minimal preprocessing is required before conducting the business analysis.